drivers/hwinfo/hwinfo_gecko.c

```diff
@@ -21,3 +23,121 @@ ssize_t z_impl_hwinfo_get_device_id(uint8_t *buffer, size_t length)
 
 	return length;
 }
+
+int z_impl_hwinfo_get_reset_cause(uint32_t *cause)
+{
+	uint32_t flags = 0;
+	uint32_t rmu_flags = RMU_ResetCauseGet();
```

Comment:
If I am understanding the EFM32 docs correctly then this SoC is unique in that reset bits accumulate between successive resets and must be cleared by explicitly calling `RMU_ResetCauseClear`. This is not what you're doing here. If the end user doesn't call `hwinfo_clear_reset_cause` which they may not then the current reset bits will show up after the next reset.

My suggestion is that that first time `hwinfo_get_reset_cause` is called you fetch the bits, and store them globally, then call `RMU_ResetCauseClear` to clear the hardware bits. On ever subsequent call to `hwinfo_get_reset_cause` return the stored value. A call to `hwinfo_clear_reset_cause` should clear this global value instead.

Reply:
From what I can tell, that is similar as to how some of the NXP Kinetis SoC do this (sticky reset causes, see https://github.com/zephyrproject-rtos/zephyr/pull/36053). Why would we want to "shield" this HW feature from the user?

Reply:
As far as I know, this is the desired behavior - the bits are kept in the reset cause register until explicitly cleared. This is how it's done in `stm32` and `nrf`. Otherwise, why would you need a `clean` function? A variable on the RAM would be cleared on a reset anyway.

Reply:
My concern is that other platforms don't behave like this and the user would rightly expect our API to behave consistently so their code is portable. The API returns the *most recent* reset cause. 

Alternative is to update the API docs to reflect that on some platforms the bits can be sticky and that the user should call  `hwinfo_clear_reset_cause` right after fetching the reason to ensure the API does what it is supposed to do -- which is give you the most recent reset cause. We'd be asking the user to do would should do internally (IMHO). Passing that to the user complicates the usage.

Reply:
The API documentation for `hwinfo_get_reset_cause()` does not mention anything about it only returning the most recent reset cause. If it should behave like that, what would is the purpose of `hwinfo_clear_reset_cause()`?

Reply:
@yonsch  - You are correct about the other platforms. It has been a while since I wrote the reset cause API and I was confused about how it should function. I withdraw this change.

Reply:
I'll update the API docs to clarify the usage.

Reply:
See #36108